Src/Particle/AMReX_ParticleUtil.H

```diff
@@ -293,6 +294,44 @@ numParticlesOutOfRange (Iterator const& pti, IntVect nGrow)
     return hv;
 }
 
+template <class Iterator, std::enable_if_t<IsParticleIterator<Iterator>::value && Iterator::ContainerType::ParticleType::is_soa_particle, int> foo = 0>
+int
+numParticlesOutOfRange (Iterator const& pti, IntVect nGrow)
+{
+    using ParticleType = typename Iterator::ContainerType::ConstParticleType;
+
+    const auto tile = pti.GetParticleTile();
```

Comment:
CI says:
```
/home/runner/work/amrex/amrex/Src/Particle/AMReX_ParticleUtil.H:303:16: error: the const qualified variable 'tile' is copy-constructed from a const reference; consider making it a const reference [performance-unnecessary-copy-initialization,-warnings-as-errors]
Suppressed 38940 warnings (38923 in non-user code, 17 NOLINT).
    const auto tile = pti.GetParticleTile();
               ^
              &
```
So let's do
```suggestion
    const auto & tile = pti.GetParticleTile();
```
?